controllers/scopeinstance_controller.go

```diff
@@ -104,210 +131,167 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// create required roleBindings and clusterRoleBindings.
 	if err := r.ensureBindings(ctx, in, st); err != nil {
-		log.Log.Error(err, "in creating Role Bindings")
+		log.Log.Error(err, "in creating RoleBindings")
 		return ctrl.Result{}, err
 	}
 
-	listOption := client.MatchingLabels{
-		scopeInstanceUIDKey: string(in.GetUID()),
-	}
-
-	requirement, err := labels.NewRequirement(scopeInstanceHashKey, selection.NotEquals, []string{util.HashObject(in.Spec)})
-	if err != nil {
+	// delete out of date (Cluster)RoleBindings
+	if err := r.deleteOldBindings(ctx, in, st); err != nil {
+		log.Log.Error(err, "in deleting (Cluster)RoleBindings")
 		return ctrl.Result{}, err
 	}
 
-	listOptions := &client.ListOptions{
-		LabelSelector: labels.NewSelector().Add(*requirement),
-	}
+	return ctrl.Result{}, nil
+}
 
-	if err := r.deleteBindings(ctx, listOption, listOptions); err != nil {
-		log.Log.Error(err, "in deleting Role Bindings")
-		return ctrl.Result{}, err
+// ensureBindings will ensure that the proper bindings are created for a given ScopeInstance and ScopeTemplate.
+// If the ScopeInstance.Spec.Namespaces is empty it will create a ClusterRoleBinding.
+// If the ScopeInstance.Spec.Namespaces is not empty it will create a RoleBinding in each provided namespace.
+// A separate (Cluster)RoleBinding will be created for each ClusterRole specified in the ScopeTemplate
+func (r *ScopeInstanceReconciler) ensureBindings(ctx context.Context, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) error {
+	for _, cr := range st.Spec.ClusterRoles {
+		if len(in.Spec.Namespaces) == 0 {
+			err := r.createOrUpdateClusterRoleBinding(ctx, &cr, in, st)
+			if err != nil {
+				return err
+			}
+		} else {
+			for _, ns := range in.Spec.Namespaces {
+				err := r.createOrUpdateRoleBinding(ctx, &cr, in, st, ns)
+				if err != nil {
+					return err
+				}
+			}
+		}
 	}
 
-	// TODO: Find out how to merge with the above delete
-	listOption = client.MatchingLabels{
-		scopeInstanceUIDKey: string(in.GetUID()),
-		scopeTemplateUIDKey: string(st.GetUID()),
-	}
+	return nil
+}
 
-	requirement, err = labels.NewRequirement(scopeTemplateHashKey, selection.NotEquals, []string{util.HashObject(st.Spec)})
-	if err != nil {
-		return ctrl.Result{}, err
+func (r *ScopeInstanceReconciler) createOrUpdateClusterRoleBinding(ctx context.Context, cr *operatorsv1.ClusterRoleTemplate, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate) error {
+	crb := r.getClusterRoleBinding(cr, in, st)
+	crbList := &rbacv1.ClusterRoleBindingList{}
+	if err := r.Client.List(ctx, crbList, client.MatchingLabels{
+		scopeInstanceUIDKey:           string(in.GetUID()),
+		clusterRoleBindingGenerateKey: cr.GenerateName,
+	}); err != nil {
+		return err
 	}
 
-	listOptions = &client.ListOptions{
-		LabelSelector: labels.NewSelector().Add(*requirement),
+	if len(crbList.Items) > 1 {
+		return fmt.Errorf("more than one ClusterRoleBinding found for ClusterRole %s", cr.GenerateName)
```

Comment:
commit https://github.com/operator-framework/oria-operator/pull/48/commits/5d6a4f9de2ac93bc1bd977eec67a61e002fe7a75

Reply:
Oops. ignore ^, wrong comment

Reply:
So we also discussed this during scrum and talked about making it so that the names for the `(Cluster)Role(Binding)`s that are created by the oria-operator use deterministic names rather than generated names and labels for finding ones associated with a particular `ScopeInstance`. The plan is to do this in a follow-up issue/PR.